packages/react-composites/src/composites/ChatComposite/ChatScreen.tsx

```diff
@@ -194,6 +196,18 @@ export const ChatScreen = (props: ChatScreenProps): JSX.Element => {
     [fileSharing?.downloadHandler]
   );
 
+  /* @conditional-compile-remove(teams-inline-images) */
+  const onRenderInlineAttachment = useCallback(
+    async (attachment: FileMetadata): Promise<AttachmentDownloadResult | undefined> => {
+      if (adapter.downloadAuthenticatedAttachment && attachment.previewUrl) {
+        const blob = await adapter.downloadAuthenticatedAttachment(attachment.previewUrl);
```

Comment:
is this cached anywhere? Otherwise we are going to call download on every time it tries to render the attachment

Reply:
Yes this is a good call out. Right now in the `MessageThread.tsx` we have a check that a specific `attachment.id` isn't already there, if it is already there we don't other updating the state so it shouldn't kick off a new re-render, so it also shouldn't call download every time either. 
We have plans to make this logic a little smarter too. Coming soon...
```js
  /* @conditional-compile-remove(teams-inline-images) */
  const [inlineAttachments, setInlineAttachments] = useState<Record<string, string>>({});
  /* @conditional-compile-remove(teams-inline-images) */
  const onFetchInlineAttachment = useCallback(
    async (attachment: FileMetadata): Promise<void> => {
      if (!onFetchAttachments || attachment.id in inlineAttachments) {
        return;
      }
      const attachmentDownloadResult = await onFetchAttachments(attachment);
      if (attachmentDownloadResult) {
        setInlineAttachments((prev) => ({ ...prev, [attachment.id]: attachmentDownloadResult.blobUrl }));
      }
    },
    [inlineAttachments, onFetchAttachments]
  );
```